contracts/Blockbin.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.4.4;
+
+contract Blockbin {
+
+    // TODO: switch to governance-style deletion instead of
+    // monarchy-style with an array of admins.
+    address BlockbinAdmin;
+
+    // TODO: do we need other fields?
+    struct Cube {
+        bytes32 hash;
+        address owner;
+        bytes data;
+    }
+
+    // Main Storage structure for Cubes
+    mapping (bytes32 => Cube) allBlockbins;
+
+
+    modifier onlyAdmin() {
+        if (msg.sender != BlockbinAdmin) {
+            revert();
+        }
+        // Do not forget the "_;"! It will be replaced by the actual function
+        // body when the modifier is used.
+        _;
+    }
+
+    function empty(bytes32 hash) returns (bool success) {
+        Cube storage cube = allBlockbins[hash];
+        if (msg.sender != cube.owner) {
+            revert();
+        }
+
+        delete allBlockbins[hash];
+        return true;
+    }
+
+    function forceEmpty (bytes32 hash) onlyAdmin returns (bool success) {
+        delete allBlockbins[hash];
+        return true;
+    }
+
+    function dumpCube(bytes data) returns (bool success) {
```

Comment:
bug! what if multiple transactions occur in the same block?

Reply:
Yeah that's a tough one. In solidity there's not good way to get the transaction nonce: http://solidity.readthedocs.io/en/develop/units-and-global-variables.html#block-and-transaction-properties

We could do `sha3(data)` to approach something unique but then 2 people wouldn't be able to dump the same content in the same transaction.

Reply:
After talking, we decided to let this go and generate this hash on the client-side. This also lets users choose arbitrary hashes ("named" cubes).